lib/Conversion/TritonGPUToLLVM/GatherOpToLLVM.cpp

```diff
@@ -0,0 +1,109 @@
+#include "triton/Conversion/TritonGPUToLLVM/PatternTritonGPUOpToLLVM.h"
+#include "triton/Conversion/TritonGPUToLLVM/Utility.h"
+
+using namespace mlir;
+using namespace mlir::triton;
+
+namespace {
+class GatherOpConversion : public ConvertOpToLLVMPattern<GatherOp> {
+public:
+  GatherOpConversion(LLVMTypeConverter &typeConverter,
+                     const TargetInfoBase &targetInfo, PatternBenefit benefit)
+      : ConvertOpToLLVMPattern(typeConverter, benefit), targetInfo(targetInfo) {
+  }
+
+  LogicalResult
+  matchAndRewrite(GatherOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override;
+
+private:
+  const TargetInfoBase &targetInfo;
+};
+
+LogicalResult
+GatherOpConversion::matchAndRewrite(GatherOp op, OpAdaptor adaptor,
+                                    ConversionPatternRewriter &rewriter) const {
+  Location loc = op.getLoc();
```

Comment:
Should there be a `isSupportedLayout` check? e.g., does it work for `DotOperandLayout`?

Reply:
I can double check, but I'm pretty sure all the code here is layout generic. Later I will add support for optimized lowerings given specific layouts.

Reply:
Since the current implementation just stores src into shared memory and `emitIndices` is generic over the layout (everything is converted to a LinearLayout), the code here works with the stranger layouts. I added another test just to make sure.